graphql_compiler/macros/__init__.py

```diff
@@ -176,7 +181,16 @@ def get_schema_for_macro_definition(schema):
     Returns:
         GraphQLSchema usable for writing macros
     """
-    raise NotImplementedError()  # TODO(bojanserafimov): Implement
+
+    required_macro_directives = (MacroEdgeDefinitionDirective, MacroEdgeTargetDirective)
+
+    # pylint: disable=protected-access
+    schema_for_macro_definition = copy(schema)
+    schema_for_macro_definition._directives = list(chain(
+        schema_for_macro_definition._directives, required_macro_directives))
```

Comment:
Can we also not include the output and output_source directives?

Reply:
Good point. Added in commit "Finish adding sets ..."